src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -993,26 +987,63 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
                 "Response version " + responseVersion);
         }
 
-        private async Task<Version> SendRequestAndGetResponseVersionAsync(Version requestVersion, Uri server)
+        private async Task<Version> SendRequestAndGetResponseVersionAsync(Version requestVersion, Uri serverUri)
         {
-            var request = new HttpRequestMessage(HttpMethod.Get, server);
-            if (requestVersion != null)
-            {
-                request.Version = requestVersion;
-            }
-            else
+            Version responseVersion = null;
+
+            if (requestVersion == null)
             {
                 // The default value for HttpRequestMessage.Version is Version(1,1).
                 // So, we need to set something different to test the "unknown" version.
-                request.Version = new Version(0,0);
+                requestVersion = new Version(0,0);
             }
+            _output.WriteLine("Request version: {0}", requestVersion);
 
-            using (var client = new HttpClient())
-            using (HttpResponseMessage response = await client.SendAsync(request))
+            if (serverUri == null)
             {
-                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-                return response.Version;
+                await LoopbackServer.CreateClientAndServerAsync(async (handler, client, server, url) =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, url);
+                    request.Version = requestVersion;
+
+                    Task<HttpResponseMessage> getResponse = client.SendAsync(request);
+
+                    await LoopbackServer.AcceptSocketAsync(server, async (s, stream, reader, writer) =>
+                    {
+                        // Server will echo back same minor HTTP/1.x version as request.
+                        string version = (request.Version == new Version(1,0)) ? "1.0" : "1.1";
```

Comment:
No.  Because then it would end up using HTTP/2.0 when the request.Version is (2,0) and that is not correct.  Only HTTP/1.0 or HTTP/1.1 appear on the status line.  HTTP/2.0 is a transport level thing and doesn't appear on the wire on the status line.